src/ray/ray_exported_symbols.lds

```diff
@@ -35,5 +34,3 @@
 *Java*
 *JNI_*
```

Comment:
do we know what symbols are exported through `*PyInit*`, `*Java*` and `*JINI_*`?

Reply:
For Python, It's better to expose specific symbols instead of general patterns, e.g. use `PyInit__raylet` instead of `*PyInit*`. But for Java, there are many symbols need to be exposed, maybe we can still use `*Java*` and `*JNI*`? @jovany-wang

Reply:
> What's the down side of being very specific instead of exposing those general patterns? i.e. instead of doing: `*PyInit*`, `*Java*` and `*JINI_*`? can we do `* PyInit__raylet*`, `*Java_ray*`...?

It's hard to say we'd list all of symbols specifically. 
Generally speaking, they are equivalent. Nothing else will be here but ray-self. If this symbol is not used internally, the compiler will automatically optimize it. Actually, this is also a common way.

Reply:
I think it's OK to add some specific suffix for Java. 
@ashione Do you have any concern?